src/displayapp/screens/Dice.cpp

```diff
@@ -0,0 +1,199 @@
+                                      0,
+                                      0);
+
+  lv_obj_t* dCounterLabel = MakeLabel(&jetbrains_mono_bold_20,
+                                      LV_COLOR_WHITE,
+                                      LV_LABEL_LONG_EXPAND,
+                                      0,
+                                      LV_LABEL_ALIGN_CENTER,
+                                      "sides",
+                                      nCounterLabel,
+                                      LV_ALIGN_OUT_RIGHT_MID,
+                                      20,
+                                      0);
+
+  nCounter.Create();
+  lv_obj_align(nCounter.GetObject(), nCounterLabel, LV_ALIGN_OUT_BOTTOM_MID, 0, 10);
+  nCounter.SetValue(1);
+
+  dCounter.Create();
+  lv_obj_align(dCounter.GetObject(), dCounterLabel, LV_ALIGN_OUT_BOTTOM_MID, 0, 10);
+  dCounter.SetValue(6);
+
+  std::uniform_int_distribution<> distrib(0, resultColors.size() - 1);
+  currentColorIndex = distrib(gen);
+
+  resultTotalLabel = MakeLabel(&jetbrains_mono_42,
+                               resultColors[currentColorIndex],
+                               LV_LABEL_LONG_BREAK,
+                               120,
+                               LV_LABEL_ALIGN_CENTER,
+                               "",
+                               lv_scr_act(),
+                               LV_ALIGN_IN_TOP_RIGHT,
+                               11,
+                               38);
+  resultIndividualLabel = MakeLabel(&jetbrains_mono_bold_20,
+                                    resultColors[currentColorIndex],
+                                    LV_LABEL_LONG_BREAK,
+                                    90,
+                                    LV_LABEL_ALIGN_CENTER,
+                                    "",
+                                    resultTotalLabel,
+                                    LV_ALIGN_OUT_BOTTOM_MID,
+                                    0,
+                                    10);
+
+  Roll();
+  openingRoll = false;
+
+  btnRoll = lv_btn_create(lv_scr_act(), nullptr);
+  btnRoll->user_data = this;
+  lv_obj_set_event_cb(btnRoll, btnRollEventHandler);
+  lv_obj_set_size(btnRoll, 240, 50);
+  lv_obj_align(btnRoll, lv_scr_act(), LV_ALIGN_IN_BOTTOM_MID, 0, 0);
+
+  btnRollLabel = MakeLabel(&jetbrains_mono_bold_20,
+                           LV_COLOR_WHITE,
+                           LV_LABEL_LONG_EXPAND,
+                           0,
+                           LV_LABEL_ALIGN_CENTER,
+                           Symbols::dice,
+                           btnRoll,
+                           LV_ALIGN_CENTER,
+                           0,
+                           0);
+
+  // Spagetti code in motion controller: it only updates the shake speed when shake to wake is on...
+  enableShakeForDice = !settingsController.isWakeUpModeOn(Pinetime::Controllers::Settings::WakeUpMode::Shake);
+  if (enableShakeForDice)
```

Comment:
You should put braces around this if-statement.

Reply:
Great point, thank you -- corrected in https://github.com/InfiniTimeOrg/InfiniTime/pull/1326/commits/d4c69757e4f8be9e3b5059cd38f2f8a90a4ba5fa